udp/session.go

```diff
@@ -41,6 +41,10 @@ func NewSession(
 	}
 }
 
+func (s *Session) SetContextValue(key interface{}, val interface{}) {
+	s.ctx = context.WithValue(s.ctx, key, val)
```

Comment:
Before `s.ctx` variable points to same address for whole time, with the new feature we need to guarded with lock.
eg `s.ctx` as atomic.Value